app/src/userpages/components/ProfilePage/IdentityHandler/AddIdentityDialog/index.jsx

```diff
@@ -1,111 +1,120 @@
 // @flow
 
-import React from 'react'
-import { connect } from 'react-redux'
+import React, { useState, useCallback, useEffect } from 'react'
+import { I18n } from 'react-redux-i18n'
 
-import withWeb3 from '$shared/utils/withWeb3'
-import { createIdentity } from '$shared/modules/integrationKey/actions'
-import type { ApiResult } from '$shared/flowtype/common-types'
-import type { IntegrationKey, IntegrationKeyList } from '$shared/flowtype/integration-key-types'
-import { areAddressesEqual } from '$mp/utils/smartContract'
-import { selectAccountId } from '$mp/modules/web3/selectors'
-import { selectEthereumIdentities } from '$shared/modules/integrationKey/selectors'
-import type { StoreState } from '$shared/flowtype/store-state'
+import useWeb3Status from '$shared/hooks/useWeb3Status'
+import Web3ErrorDialog from '$shared/components/Web3ErrorDialog'
+import useEthereumIdentities from '$shared/modules/integrationKey/hooks/useEthereumIdentities'
+import useModal from '$shared/hooks/useModal'
 import type { Address } from '$shared/flowtype/web3-types'
+import UnlockWalletDialog from '$mp/components/Modal/UnlockWalletDialog'
+import { areAddressesEqual } from '$mp/utils/smartContract'
 
 import IdentityNameDialog from '../IdentityNameDialog'
 import IdentityChallengeDialog from '../IdentityChallengeDialog'
 import DuplicateIdentityDialog from '../IdentityChallengeDialog/DuplicateIdentityDialog'
 
-type OwnProps = {
-    onClose: () => void,
-}
-
-type StateProps = {
-    accountId: ?Address,
-    ethereumIdentities: ?IntegrationKeyList,
-}
-
-type DispatchProps = {
-    createIdentity: (name: string) => ApiResult<IntegrationKey>,
+type Props = {
+    api: Object,
+    requiredAddress?: Address,
 }
 
-type Props = OwnProps & StateProps & DispatchProps
-
 const identityPhases = {
     NAME: 'name',
     CHALLENGE: 'challenge',
     COMPLETE: 'complete',
 }
 
-type State = {
-    phase: $Values<typeof identityPhases>,
-}
+const AddIdentityDialog = ({ api, requiredAddress }: Props) => {
+    const { web3Error, checkingWeb3, account } = useWeb3Status()
```

Comment:
Just a thought, maybe `account` to something like `web3Account` since there's some ambiguity with 'user account' throughout the app. Bit of a refactor job though.

Reply:
I think it's ok to have it like that in the hook, you can always rename it:
```
const { account: web3Account } = useWeb3Status()
```